### PR TITLE
Fix libafl_qemu i386 build

### LINKED ISSUE #918: some bugs in libafl_qemu i386
I just try to follow the [Hacking TMNF - Fuzzing the game server](https://blog.bricked.tech/posts/tmnf/part1/) tutorial, i git clone it's source code and then run ```CROSS_CC="gcc" cargo run```, finally i got some errors.

**To Reproduce**
1. write ```libafl_qemu = { path = "../LibAFL/libafl_qemu/", features = ["i386", "usermode"] }``` in Cargo.toml (but there are no problem with "x86_64" )

**Expected behavior**
```cargo run ``` executed successfully

**Screen output/Screenshots**
图片

**Additional context**
I spend some time auditing code, i found two problems:
1. in ```libafl_qemu/src/snapshot.rs: line 21```, use SYS_newfstatat, but i386 arch do not have syscall SYS_newfstatat
2. in ```libafl_qemu/src/block.rs: line 46```, the absence of cpu_target="i386" condition lead to cs not found in this scope error

these bug looks strange, i'm not sure if LibAFL_qemu not support i386 arch anymore or just forget consider this situation when coding.

### PR DESCRIPTION
**Pull request by tokatoka**
Fix #918 

@andreafioraldi 
Please check this

### CODE REVIEW

File: libafl_qemu/src/snapshot.rs [hunk 1]
```diff
@@ -17,7 +17,11 @@ use crate::{
 };
 #[cfg(cpu_target = "arm")]
 use crate::{SYS_fstatat64, SYS_mmap2};
-#[cfg(not(cpu_target = "arm"))]
+
+#[cfg(cpu_target = "i386")]
+use crate::{SYS_fstatat64, SYS_mmap};
```
Comment (tokatoka) [inline libafl_qemu/src/snapshot.rs:22]: newfstatat is not available on i386 so use fstatat64 like arm here

File: libafl_qemu/src/snapshot.rs [hunk 2]
```diff
@@ -15,9 +15,11 @@ use crate::{
     GuestAddr, SYS_fstat, SYS_fstatfs, SYS_futex, SYS_getrandom, SYS_mprotect, SYS_mremap,
     SYS_munmap, SYS_pread64, SYS_read, SYS_readlinkat, SYS_statfs,
 };
+#[cfg(cpu_target = "i386")]
+use crate::{SYS_fstatat64, SYS_mmap};
```
Comment (domenukk) [inline libafl_qemu/src/snapshot.rs:19]: Why SYS_mmap here but `not` it below? Maybe split this up into fstatat and map, per arch

**Comment by domenukk** [comment]
Should this be caught in CI?

**Comment by tokatoka** [comment]
the problem is libafl_qemu does not test for all architectures

**Pull request merged**